src/parser/view_transform.c

```diff
@@ -3725,6 +3726,48 @@ pt_check_pushable_term (PARSER_CONTEXT * parser, PT_NODE * term, FIND_ID_INFO *
   return PT_PUSHABLE_TERM (infop) && !is_correlated_with_agg && !is_correlated_with_dblink;
 }
 
+/*
+ * pt_copy_term_for_dblink () - copies exactly a term node excluding cast node for dblink,
+      and returns a pointer to the copy. It is eligible for a walk "pre" function
+ *   return:
+ *   parser(in):
+ *   old_node(in):
+ *   arg(in):
+ *   continue_walk(in):
+ */
+static PT_NODE *
+pt_copy_term_for_dblink (PARSER_CONTEXT * parser, PT_NODE * old_node, void *arg, int *continue_walk)
+{
+  PT_NODE *new_node = NULL;
+
+  if (old_node->node_type == PT_EXPR)
+    {
+      if (old_node->info.expr.op == PT_CAST && PT_EXPR_INFO_IS_FLAGED (old_node, PT_EXPR_INFO_CAST_WRAP))
+	{
+	  new_node = parser_new_node (parser, old_node->info.expr.arg1->node_type);
+	  if (new_node == NULL)
+	    {
+	      PT_INTERNAL_ERROR (parser, "allocate new node");
+	      return NULL;
+	    }
+	  *new_node = *old_node->info.expr.arg1;
```

Comment:
nvl(col1) in cast( nvl(col1) as int) cannot be copied. It is recommended to make a full copy and remove cast().

Reply:
old_node (cast () ) is needed to free.

Reply:
please show the full query as a incorrect rewrite result.

Reply:
I added the routin "pt_remove_cast_wrap_for_dblink"